### PR TITLE
Rename TestEnableDevelopmentHardForkEras and TestEnableDevelopmentNetworkProtocols

### PR DESCRIPTION
**Pull request by LudvikGalois**
`TestEnableDevelopmentHardForkEras` has been renamed to
`ExperimentalHardForksEnabled `. An error is thrown if
`TestEnableDevelopmentHardForkEras` is used to avoid it silently being set
to `False`.

`TestEnableDevelopmentNetworkProtocols` has been renamed to
`ExperimentalProtocolsEnabled`. An error is thrown if
`TestEnableDevelopmentNetworkProtocols` is used to avoid it silently being set
to `False`.

Closes https://github.com/input-output-hk/cardano-node/issues/4043
Closes https://github.com/input-output-hk/cardano-node/issues/4470

### CODE REVIEW

**Comment by newhoggy** [comment]
It won't be enough to just change the code.   There are some tests that use the `TestEnableDevelopmentHardForkEras` configuration that now need to do something else.  Grep over the code to find these occurrences.

**Comment by LudvikGalois** [comment]
> It won't be enough to just change the code. There are some tests that use the `TestEnableDevelopmentHardForkEras` configuration that now need to do something else. Grep over the code to find these occurrences.

Fixed

File: cardano-node/src/Cardano/Node/Configuration/POM.hs [hunk 1]
```diff
@@ -374,8 +385,16 @@ instance FromJSON PartialNodeConfiguration where
              }
 
       parseHardForkProtocol v = do
-        npcTestEnableDevelopmentHardForkEras
-          <- v .:? "TestEnableDevelopmentHardForkEras"
+
+        failOnRemovedFields v
+          [ RemovedField
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Node/Configuration/POM.hs:390]: :+1:

File: cardano-node/src/Cardano/Node/Protocol/Cardano.hs [hunk 2]
```diff
@@ -85,12 +85,6 @@ mkSomeConsensusProtocolCardano NodeByronProtocolConfiguration {
                              npcAlonzoGenesisFileHash
                            }
                            NodeHardForkProtocolConfiguration {
-                            -- npcTestEnableDevelopmentHardForkEras,
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Node/Protocol/Cardano.hs:88]: So this comment is still valid. We only need to reflect the name change in the comment.

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]
One minor change

File: cardano-node/src/Cardano/Node/Protocol/Cardano.hs (hunk 2 above)
Comment (Jimbo4350) [reply cardano-node/src/Cardano/Node/Protocol/Cardano.hs:88]: @LudvikGalois this is not resolved. The comment is still deleted.

Comment (LudvikGalois) [reply cardano-node/src/Cardano/Node/Protocol/Cardano.hs:88]: Because the field has been renamed. A new comment was added

Comment (Jimbo4350) [reply cardano-node/src/Cardano/Node/Protocol/Cardano.hs:88]: Where is the new comment?

Comment (LudvikGalois) [reply cardano-node/src/Cardano/Node/Protocol/Cardano.hs:88]: On line 88, where the old one was https://github.com/input-output-hk/cardano-node/pull/4341/files#diff-8c649864fe4b96d142cfd4c2bb9e32a9510fe737c6ad4e13e04b585ee15a6b74R88

Comment (Jimbo4350) [reply cardano-node/src/Cardano/Node/Protocol/Cardano.hs:88]: The comment being deleted provides context as to what the flag is being used for. Without it newcomers will have a more difficult time understanding the purpose of the flag.

Comment (Jimbo4350) [reply cardano-node/src/Cardano/Node/Protocol/Cardano.hs:88]: `--npcTestEnableDevelopmentProtVer` is not a sufficient replacement for the comment being deleted. Please restore the comment.

Comment (LudvikGalois) [reply cardano-node/src/Cardano/Node/Protocol/Cardano.hs:88]: My mistake, I completely misunderstood what was meant by "the comment"

**Comment by newhoggy** [comment]
In the diff, I see renames to `TestEnableAdvertiseDevelopmentProtVer`, but in the PR description, and change log it notes renames to `TestEnableDevelopmentProtVer`.

Can I confirm the code changes are correct and the change log and PR description need to be updated to match the code?

**Review by Jimbo4350 — APPROVED** [review]
LGTM!

**Review by disassembler — APPROVED** [review]

**Pull request merged**